google-cloud-bigquery/lib/google/cloud/bigquery/encryption.rb

```diff
@@ -0,0 +1,123 @@
+      # arguments to {Dataset#create_table} and
+      # {Table#encryption_configuration}.
+      #
+      # @see https://cloud.google.com/bigquery/docs/customer-managed-encryption
+      #   Protecting Data with Cloud KMS Keys
+      #
+      # @example
+      #   require "google/cloud/bigquery"
+      #
+      #   bigquery = Google::Cloud::Bigquery.new
+      #   dataset = bigquery.dataset "my_dataset"
+      #   encrypt_config = Google::Cloud::Bigquery::EncryptionConfiguration.new
+      #   encrypt_config.kms_key_name = "my_key_name"
+      #   table = dataset.create_table "my_table",
+      #                                encryption_configuration: encrypt_config
+      #
+      class EncryptionConfiguration
+        ##
+        # @private The Google API Client object.
+        attr_accessor :gapi
+
+        ##
+        # @private Create an empty EncryptionConfiguration object.
+        def initialize
+          @gapi = Google::Apis::BigqueryV2::EncryptionConfiguration.new
+        end
+
+        ##
+        # The Cloud KMS encryption key that will be used to protect the table.
+        # For example: `projects/a/locations/b/keyRings/c/cryptoKeys/d`
+        # The default value is `nil`, which means default encryption is used.
+        #
+        # @return [String]
+        #
+        # @example
+        #   require "google/cloud/bigquery"
+        #
+        #   config = Google::Cloud::Bigquery::EncryptionConfiguration.new
+        #   key_name = "projects/a/locations/b/keyRings/c/cryptoKeys/d"
+        #   config.kms_key_name = key_name
+        #
+        def kms_key_name
+          @gapi.kms_key_name
+        end
+
+        ##
+        # Set the Cloud KMS encryption key that will be used to protect the
+        # table. For example: `projects/a/locations/b/keyRings/c/cryptoKeys/d`
+        # The default value is `nil`, which means default encryption is used.
+        #
+        # @param [String] new_kms_key_name New Cloud KMS key name
+        #
+        # @example
+        #   require "google/cloud/bigquery"
+        #
+        #   config = Google::Cloud::Bigquery::EncryptionConfiguration.new
+        #   key_name = "projects/a/locations/b/keyRings/c/cryptoKeys/d"
+        #   config.kms_key_name = key_name
+        #
+        def kms_key_name= new_kms_key_name
+          frozen_check!
+          @gapi.kms_key_name = new_kms_key_name
+        end
+
+        # @private
+        def changed?
+          return false if frozen?
+          @original_json != @gapi.to_json
+        end
+
+        ##
+        # @private Google API Client object.
+        def to_gapi
+          @gapi
+        end
+
+        ##
+        # @private Google API Client object.
+        def self.from_gapi gapi
+          new_config = new
+          new_config.instance_variable_set :@gapi, gapi
+          new_config
+        end
+
+        # @private
+        def == other
+          return false unless other.is_a? Schema
```

Comment:
Is this supposed to be:

```ruby
return false unless other.is_a? EncryptionConfiguration
```

Reply:
Ah! That explains why I couldn't get equality to work in my tests.